### PR TITLE
doxygen fixes for sorc/grid_tools.fd

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #191 
Part of #343 

More doxygen fixes. I am working with the code author to fill in the ???s, but this PR will silence a bunch of doxygen warnings.

### CODE REVIEW

File: sorc/grid_tools.fd/regional_esg_grid.fd/regional_esg_grid.f90 [hunk 1]
```diff
@@ -2,7 +2,7 @@
 !! @brief ???
```
Comment (GeorgeGayno-NOAA) [inline sorc/grid_tools.fd/regional_esg_grid.fd/regional_esg_grid.f90:2]: I can fill in this ???

File: sorc/grid_tools.fd/regional_esg_grid.fd/regional_esg_grid.f90 [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
 !! @author R. J. Purser
 
-!! Program which does ???
+!> Program which does ???
```
Comment (GeorgeGayno-NOAA) [inline sorc/grid_tools.fd/regional_esg_grid.fd/regional_esg_grid.f90:5]: I can fill in this ???

File: sorc/grid_tools.fd/filter_topo.fd/filter_topo.F90 [hunk 3]
```diff
@@ -1,4 +1,11 @@
 !> @file
+!! @brief This program does ???
+!! @author R. J. Purser
+
+!> This program does ???
+!!
+!! @return 0 for success, error code otherwise.
+!! @author R. J. Purser
```
[the remark this answers is not in the document]
Comment (edwardhartnett) [reply sorc/grid_tools.fd/filter_topo.fd/filter_topo.F90]: OK, I will ask Rusty.

File: sorc/grid_tools.fd/global_equiv_resol.fd/global_equiv_resol.f90 [hunk 4]
```diff
@@ -1,7 +1,12 @@
 !> @file
-!=======================================================================
+!! @brief This program does ???
+!! @author R. J. Purser
+
+!> This program does ???
+!!
+!! @author R. J. Purser
```
[the remark this answers is not in the document]
Comment (GeorgeGayno-NOAA) [reply sorc/grid_tools.fd/global_equiv_resol.fd/global_equiv_resol.f90]: Updated.

File: sorc/grid_tools.fd/regional_esg_grid.fd/regional_esg_grid.f90 (hunk 1 above)
Comment (edwardhartnett) [reply sorc/grid_tools.fd/regional_esg_grid.fd/regional_esg_grid.f90:2]: OK, check out the branch or just type it here...

**Pull request merged**